src/main/scala/akka/persistence/jdbc/journal/dao/ByteArrayJournalDao.scala

```diff
@@ -85,12 +85,8 @@ trait BaseByteArrayJournalDao extends JournalDaoWithUpdates {
   }
 
   private def writeJournalRows(xs: Seq[JournalRow]): Future[Unit] = {
-    if (slickProfile != H2Profile)
-      // Write atomically without auto-commit
-      db.run(queries.writeJournalRows(xs).transactionally).map(_ => Unit)
-    else
-      // However transactionally causes H2 tests to fail
-      db.run(queries.writeJournalRows(xs)).map(_ => Unit)
+    // Write atomically without auto-commit
+    db.run(queries.writeJournalRows(xs).transactionally).map(_ => Unit)
```

Comment:
I found that I got circuit breaker timeouts in the H2 tests, specifically `H2ScalaCurrentEventsByTagTest`, "complete without any gaps in case events are being persisted when the query is executed" where it tries to write 10,000 messages. For example see [this build](https://travis-ci.org/jonfox/akka-persistence-jdbc/jobs/429770945).

Also taking off `transactionally` for SQL Server caused timeouts, although I didn't have Travis hooked up at that time. It looked like this was down to each insert being auto committed rather than as a single transaction.

Reply:
Ok, got it. 

I think it will be better if we find the cause of it. We can leave the `transactionally`, but better without the if/else. It seems that we are tweaking the code to make Travis happy while there is no functionality requiring one or the other. 

Let's see how the build will behave here.

Reply:
An alternative could be to configure a smaller `batchSize` for h2, this might also make a difference

Reply:
I'm guessing this is down to implementation differences in the Slick profiles, which this is effectively working around, i.e. what does `JournalTableC ++= xs.sortBy(_.sequenceNumber)` actually do for each profile. It seems that `++=` inserts sequentially with auto commit at least for the SQL Server profile in the absence of `transactionally`.

At least [here](http://slick.lightbend.com/doc/3.2.3/dbio.html) under "Transactions and Pinned Sessions" it seems to imply that we are only guaranteed to be run in a transaction if `transactionally` is used.

It would be useful to know whether `++-` in Slick is supposed to guarantee atomicity or not, and of course why H2 under performs when `transactionally` is used.

Reply:
Yeah, I'm surprised by this part of Slick docs:
> Without it (transactionally), all database actions run in auto-commit mode

So, let's leave the `transactionally` call and try to fix it for H2.